cql3/result_set.cc

```diff
@@ -18,14 +18,22 @@ namespace cql3 {
 metadata::metadata(std::vector<lw_shared_ptr<column_specification>> names_)
         : _flags(flag_enum_set())
         , _column_info(make_lw_shared<column_info>(std::move(names_)))
-{ }
+{
+    if (!_column_info->_names.empty() && column_specification::all_in_same_table(_column_info->_names)) {
+        _flags.set<flag::GLOBAL_TABLES_SPEC>();
+    }
+}
```

Comment:
:question: Just to make sure: is setting this flag enough for the message encoding part to really encode only a global tables spec and skip per-column table specs?

Reply:
Yes.